src/main/webapp/script.js

```diff
@@ -1,54 +1,75 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
 /**
  * Event processing for upload button
  */
-function fileUploadDisplay() {
+function openFileUploadDialog() {
   document.querySelector('.bg-model').style.display = 'flex';
 }
 
 /**
  * Event processing for close button
  */
-function closeDiv() {
+function closeFileUploadDialog() {
   document.querySelector('.bg-model').style.display = 'none';
 }
 
 /**
- * Makes the form for uploading an image visible and adds the 'action' to it by fetching
+ * Makes the form for uploading an image visible and
+ * adds the 'action' to it by fetching
  * the url(that the form needs to post to) from the servlet
  */
-function fetchBlobstoreUrlAndShowForm() {
-  fetch('/blobstore-upload-url')
-      .then((response) => response.text())
-      .then((imageUploadUrl) => {
-        const uploadForm = document.getElementById('upload-barcode-form');
-        uploadForm.action = imageUploadUrl;
-        uploadForm.classList.remove('hidden');
-      });
+async function fetchBlobstoreUrlAndShowForm() {
+  const response = await fetch('/blobstore-upload-url');
+  let imageUploadUrl;
+  if (!response.ok) {
+    throw new Error(`HTTP error! status: ${response.status}`);
+  } else {
```

Comment:
Remove the else-branch since the if-branch is throwing an exception ending the execution of the function. In general, don't use the else-branch when you end the if-branch with a `return` or `throw` statement (or `continue` and `break` for loops).

Reply:
I'll  make a note of it and remember to do so.